src/components/OptionsSelector.js

```diff
@@ -213,47 +225,101 @@ class OptionsSelector extends Component {
+        const defaultMaxParticipantsReachedMessage = _.isUndefined(this.props.maxParticipantsReachedMessage)
+            ? this.props.translate('common.maxParticipantsReached', {count: this.props.selectedOptions.length})
+            : this.props.maxParticipantsReachedMessage;
         return (
-            <View style={[styles.flex1]}>
-                <View style={[styles.ph5, styles.pv3]}>
-                    <TextInput
-                        ref={el => this.textInput = el}
-                        value={this.props.value}
-                        onChangeText={(text) => {
-                            if (this.props.shouldFocusOnSelectRow) {
-                                this.textInput.setNativeProps({selection: null});
-                            }
-                            this.props.onChangeText(text);
-                        }}
-                        onKeyPress={this.handleKeyPress}
-                        placeholder={this.props.placeholderText
-                            || this.props.translate('optionsSelector.nameEmailOrPhoneNumber')}
-                        onBlur={(e) => {
-                            if (!this.props.shouldFocusOnSelectRow) {
-                                return;
-                            }
-                            this.relatedTarget = e.relatedTarget;
-                        }}
-                        selectTextOnFocus
-                    />
-                </View>
-                <OptionsList
-                    ref={el => this.list = el}
-                    optionHoveredStyle={styles.hoveredComponentBG}
-                    onSelectRow={this.selectRow}
-                    sections={this.props.sections}
-                    focusedIndex={this.state.focusedIndex}
-                    selectedOptions={this.props.selectedOptions}
-                    canSelectMultipleOptions={this.props.canSelectMultipleOptions}
-                    hideSectionHeaders={this.props.hideSectionHeaders}
-                    headerMessage={this.props.headerMessage}
-                    disableFocusOptions={this.props.disableArrowKeysActions}
-                    hideAdditionalOptionStates={this.props.hideAdditionalOptionStates}
-                    forceTextUnreadStyle={this.props.forceTextUnreadStyle}
-                    showTitleTooltip={this.props.showTitleTooltip}
-                />
-            </View>
+            <ArrowKeyFocusManager
+                ref={el => this.focusManager = el}
+                listLength={this.props.canSelectMultipleOptions ? this.state.allOptions.length + 1 : this.state.allOptions.length}
+                onFocusedIndexChanged={this.scrollToIndex}
+                onEnterKeyPressed={this.selectFocusedIndex}
+                shouldEnterKeyEventBubble={focusedIndex => !this.state.allOptions[focusedIndex]}
+            >
+                {({focusedIndex}) => {
+                    this.focusedOption = this.state.allOptions[focusedIndex];
```

Comment:
**question:** Why are we using an instance variable for the `focusedOption`? It looks like we are just accessing it in `componentDidMount()`? Would this code have run by that point or not?

Reply:
in `componentDidMount` we set up a keyboard event listener that uses `focusedOption`. At the time that callback is registered, this code would not have run, but by the time the keyboard event is triggered by the user and the callback runs it will have run.

Reply:
Thoughts on moving the subscription to the `ArrowKeyFocusManager ` itself?

The callback here could be provided as a prop callback that returns the `focusedIndex`?

https://github.com/Expensify/App/blob/baea43efef2e1a54e21791e4edb4bc03be918000/src/components/OptionsSelector.js#L134-L140

```js
<ArrowFocusManager
    onEnterKeyPressed={(focusedIndex) => {...}}    
```

Reply:
This has changed – the enter key subscriber is still in `OptionsSelector`, but `focusedOption` is not an instance variable anymore.